readme.md

```diff
@@ -1,88 +1,183 @@
+  </a>
+</p>
 
-## Contribution
+| | Open Source version | Pro version |
+|-----|-------|-------|
+| Browser identification | Basic | Advanced |
+| Anonymous user identification | ❌ | ✅ |
+| Bot detection | ❌ | ✅ |
+| Incognito / Private mode detection | ❌ | ✅ |
+| Geolocation | ❌ | ✅ |
+| Security | ❌ | ✅ |
+| Server API | ❌ | ✅ |
+| Webhooks | ❌ | ✅ |
 
-### Development playground
+Pro result example:
 
-```bash
-# Make sure you have Yarn installed
-yarn install
-yarn playground:start # Add '--port 8765' to change the server port
+```js
+{
+  "requestId": "HFMlljrzKEiZmhUNDx7Z",
+  "visitorId": "kHqPGWS1Mj18sZFsP8Wl",
+  "visitorFound": true,
+  "incognito": false,
+  "bot": undefined,
+  "ip": "192.65.67.131",
+  "browserName": "Chrome",
+  "browserVersion": "85.0.4183",
+  "os": "Mac OS X",
+  "osVersion": "10.15.6",
+  "device": "Other",
+  "ipLocation": {/* ... */}
+}
 ```
 
-Then open http://localhost:8080 in a browser.
-It's reloaded every time the source code is changed.
+🍿 [Live demo](https://fingerprintjs.com/demo)
 
-To build the playground distribution code (e.g. to upload to a static server), run:
+⏱ [How to upgrade from Open Source to Pro in 30 seconds]()
 
-```bash
-yarn playground:build
-```
+📗 [FingerprintJS Pro documentation](https://dev.fingerprintjs.com)
+
+## Open version reference
+
+The library is shipped in various formats:
+
+- Plain JS
+    ```html
+    <script
+      async src="https://cdn.jsdelivr.net/npm/@fingerprintjs/fingerprintjs@3/dist/fp.min.js"
+      onload="/* ... */"
+    ></script>
+    ```
+- UMD
+    ```js
+    require(['https://cdn.jsdelivr.net/npm/@fingerprintjs/fingerprintjs@3/dist/fp.umd.min'], FingerprintJS => {/* ... */});
+    ```
+- EcmaScript module
+    ```js
+    import * as FingerprintJS from '@fingerprintjs/fingerprintjs';
+    ```
+- CommonJS
+    ```js
+    const FingerprintJS = require('@fingerprintjs/fingerprintjs');
+    ```
 
-The result will appear at `playground/dist`.
+API:
 
-### How to build
+- `FingerprintJS.load({ delayFallback?: number }): Promise<Agent>`
 
-To build the distribution files of the library, run:
+    Builds an instance of Agent and waits a delay required for a proper operation.
+    `delayFallback` is duration (ms) of the fallback for browsers that don't support [requestIdleCallback](https://developer.mozilla.org/en-US/docs/Web/API/Window/requestIdleCallback).
+
+- `agent.get({ debug?: boolean }): Promise<{ visitorId: string, components: {/* ... */} }>`
+
+    Gets the visitor identifier.
+    `debug: true` prints debug messages to the console.
+    `visitorId` is the visitor identifier.
+    `components` is a dictionary of components that have formed the identifier;
+    each value is an object like `{ value: any, duration: number }` in case of success
+    and `{ error: object, duration: number }` in case of an unexpected error during getting the component.
+
+- `FingerprintJS.componentsToCanonicalString(components: object): string`
```

Comment:
why do we need this function to be public? Can we have `FingerprintJS.hashComponents(components: object)` and make this function private?

Reply:
We don’t, actually publishing only a single components-to-hash function is better for us.